### ISSUE TITLE
W1b: `PadTo` silently crops a smaller target, and its `inverse` returns the cropped tensor unchanged

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4410]
Extends #3518, which reports the forward half: `PadTo((h, w))` on an input larger than `(h, w)` crops the bottom and the right instead of leaving the image alone. This issue adds the executed `inverse` half, which #3518 does not cover and which makes the data loss unrecoverable rather than merely surprising.

`PadTo.compute_transformation` returns the identity matrix — `kornia/augmentation/_2d/geometric/pad.py:72` already carries `# TODO: It is incorrect to return identity` — and `inverse` is derived from that matrix. In the pad direction the round trip works: `PadTo((10, 12))` on a 6 × 8 input inverts to the exact original. In the crop direction `inverse` neither restores the discarded rows and columns nor reports that it cannot; it returns the forward output bitwise identical. A detection pipeline that pads to a fixed size, augments, and inverts to recover original-resolution boxes gets the cropped canvas back with no signal, which is the pipeline #3518's reporter describes.

`PadTo` also mixes the two directions per axis: `PadTo((4, 12))` on a 6 × 8 input crops the height and pads the width in the same call.

## Repro

```python
import torch, kornia, kornia.augmentation as K
print("kornia", kornia.__version__, "torch", torch.__version__)
x = torch.zeros(2, 1, 6, 8); x[0, :, 1, 2] = 1.0; x[1, :, 4, 5] = 1.0

aug = K.PadTo((4, 5), pad_value=9.0)          # target SMALLER than the 6 x 8 input
o = aug(x)
inv = aug.inverse(o)                           # no exception
print("input", tuple(x.shape), "-> forward", tuple(o.shape), "-> inverse", tuple(inv.shape))
print("mass   in", x.sum().item(), " forward", o.sum().item(), " inverse", inv.sum().item())
print("inverse is bitwise the forward output:", torch.equal(inv, o))
print("transform_matrix:", aug.transform_matrix[0].tolist())

pad = K.PadTo((10, 12), pad_value=9.0)         # the pad direction round-trips exactly
p = pad(x); pinv = pad.inverse(p)
print("pad  6x8 -> ", tuple(p.shape), "-> inverse", tuple(pinv.shape),
      " max|inverse - input| =", (pinv - x).abs().max().item())
```

## Observed

```
kornia 0.9.0rc1 torch 2.14.0
input (2, 1, 6, 8) -> forward (2, 1, 4, 5) -> inverse (2, 1, 4, 5)
mass   in 2.0  forward 1.0  inverse 1.0
inverse is bitwise the forward output: True
transform_matrix: [[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]]
pad  6x8 ->  (2, 1, 10, 12) -> inverse (2, 1, 6, 8)  max|inverse - input| = 0.0
```

Sample 0's marker at `(1, 2)` survives the crop; sample 1's at `(4, 5)` is destroyed — the mass drops from `2.0` to `1.0` and `inverse` does not bring it back.

## Expected

Pad-only semantics by default: a target smaller than the input either raises, or leaves that axis untouched, with cropping available behind an explicit opt-in. PR #3964 takes that route with a `crop_if_exceeds` argument and is the natural home for the forward half.

Whichever default is chosen, `inverse` must not return its own input while presenting itself as an inverse. The two coherent outcomes there are to raise when the forward direction was lossy, or to restore the discarded region from the padding value so the shape at least round-trips — and either way `compute_transformation` should stop returning the identity for a transform that is not the identity, which is what the in-tree TODO asks for. Any change to the `inverse` contract must land in the same release as the forward-direction decision, since the two are read together.

## Classification

**Broken output** — image content is discarded with no signal, and the documented inverse silently returns the lossy output rather than undoing it. Idiom: 4 (degenerate/singular input silently producing garbage — here a target smaller than the input, which the class has no branch for).

## Disposition

`MERGE IN WINDOW — behavior change`. One issue for both halves: they share one root cause, the identity transform at `pad.py:72`, and #3964 already proposes the forward-direction argument. Extends #3518 rather than duplicating it; the new material is the executed `inverse` behavior and the `transform_matrix` literal. Batch 6 documents the crop direction and the no-op inverse on the class and pins them, and does not change either.

## Environment

kornia 0.9.0rc1 @ `d6a2c138` (main), torch 2.14.0, python 3.11, cpu. Found during the augmentation conventions audit (batch 6, tracking issue #4407); the batch documents the behavior above as it is and pins it, and does not change it.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
